ovmf/abi/pihob.go

```diff
@@ -197,3 +206,53 @@ func (d EFIHOBResourceDescriptor) WriteTo(w io.Writer) (int64, error) {
 	}
 	return SizeofEFIHOBResourceDescriptor, nil
 }
+
+// EFIHOBGUID is for an EFI_HOB_TYPE_GUID_EXTENSION for named unstructured data associated with a
+// given GUID.
+type EFIHOBGUID struct {
+	Header EFIHOBGenericHeader
+	GUID   EFIGUID
+	Data   []byte
+}
+
+// WriteTo writes the HOB GUID to the given writer.
+func (h EFIHOBGUID) WriteTo(w io.Writer) (int64, error) {
+	if h.Header.HobType != EFIHOBTypeGUIDExtension {
+		return 0, fmt.Errorf("invalid HOB type: %d", h.Header.HobType)
+	}
+	wantLength := SizeofHOBGUID + len(h.Data)
+	if int(h.Header.HobLength) != wantLength {
+		return 0, fmt.Errorf("invalid HOB length: %d, want %d", h.Header.HobLength, wantLength)
+	}
+	if n, err := h.Header.WriteTo(w); err != nil || n != SizeofHOBGenericHeader {
+		return 0, err
```

Comment:
If err is nil but n is wrong, this function will return an err with an err of nil, so the caller might think it succeeded? WDYT?

Reply:
Fixed I think?